p2p/chain.py

```diff
@@ -242,8 +255,16 @@ def _should_skip_body_download(self, header: BlockHeader) -> bool:
             key = (header.transaction_root, header.uncles_hash)
             self._pending_bodies[key] = (header, now)
 
-    def _should_skip_receipts_download(self, header: BlockHeader) -> bool:
-        return header.receipt_root == self.chaindb.empty_root_hash
+    def _skip_empty_and_duplicated_receipts(self, headers: List[BlockHeader]) -> List[BlockHeader]:
+        # Post-Byzantium blocks may have identical receipt roots (e.g. when they have the same
+        # number of transactions and all succeed/failed: ropsten blocks 2503212 and 2503284), so
+        # we have an extra check here to avoid requesting those receipts multiple times.
+        headers = list(unique(headers, key=operator.attrgetter('receipt_root')))
+        return [
```

Comment:
Same here about moving this out of a list comprehension and using `to_tuple` to allow expansion of this logic across multiple lines within a `for` loop.

Reply:
Makes sense. Doing that on #492 though as that's where it was introduced